pinot-core/src/main/java/org/apache/pinot/server/realtime/ControllerLeaderLocator.java

```diff
@@ -102,10 +102,20 @@ public static ControllerLeaderLocator getInstance() {
    * Checks whether lead controller resource has been enabled or not.
    * If yes, updates lead controller pairs from the external view of lead controller resource.
    * Otherwise, updates lead controller pairs from Helix cluster leader.
+   * Note: Exception may happen due to Helix/ZK disconnect. If so, we should NOT regress the behavior back to false.
+   * Thus, simply exiting the method should be enough. Retry will be done in the next request.
    */
   private void refreshControllerLeaderMap() {
     // Checks whether lead controller resource has been enabled or not.
-    if (LeadControllerUtils.isLeadControllerResourceEnabled(_helixManager)) {
+    boolean leadControllerResourceEnabled;
```

Comment:
I would recommend you to wrap the whole method into a try-catch because `refreshControllerLeaderMapFromHelixClusterLeader()` can also throw exception.